constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/IntervalMap.java

```diff
@@ -213,6 +235,27 @@ IntervalMap<R> map2(
       final IntervalMap<V1> left,
       final IntervalMap<V2> right,
       final BiFunction<Optional<V1>, Optional<V2>, Optional<R>> transform
+  ) {
+    return map2(left, right, (interval, l, r) -> transform.apply(l, r));
+  }
+
+  /**
+   * A generalized binary operation between two IntervalMaps.
+   *
+   * @param left left operand
+   * @param right right operand
+   * @param transform a function that transforms an interval and two {@link Optional}s of the left and right operands' types to an
+   *                  optional of a new type.
+   * @param <V1> value type of the left operand
+   * @param <V2> value type of the right operand
+   * @param <R> value type of the result
+   * @return a new IntervalMap, the result of applying the transform
+   */
+  public static <V1, V2, R>
+  IntervalMap<R> map2(
```

Comment:
I created it because I thought I had a use case when refactoring the operations on linear profiles, but when I changed it to using point-slope form I no longer needed it. So while its currently not used, its not unreasonable.

Also it mirrors the variants for `map`, which are all used.